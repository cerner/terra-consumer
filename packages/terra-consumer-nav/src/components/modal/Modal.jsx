
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import Button from 'terra-button';
import IconClose from 'terra-icon/lib/icon/IconClose';
import Overlay from 'terra-overlay';
import DangerousHtml from '../dangerous-html/DangerousHtml';
import styles from './Modal.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Title to be rendered on top of the modal content
   */
  title: PropTypes.string,
  /**
   * Boolean value to render modal.
   */
  isModalOpen: PropTypes.bool.isRequired,
  /**
   * Required callback function for use by parent component to update state.
   */
  onRequestClose: PropTypes.func.isRequired,
  /**
   * Content to be rendered in the body of the dialog box.
   */
  children: PropTypes.node,
};

const defaultProps = {
  isModalOpen: false,
};

const Modal = ({
  title,
  isModalOpen,
  onRequestClose,
  children,
  ...customProps
}) => {
  const modalHeader = (
    <div className={cx('modal-title')}>
      <DangerousHtml text={title} />
      <Button className={cx('close-button')} onClick={onRequestClose}>
        <IconClose />
      </Button>
    </div>
  );

  return (
    <Overlay {...customProps} isOpen={isModalOpen} className={cx('overlay')}>
      {modalHeader}
      <div className={cx('modal')}>
        {children}
      </div>
    </Overlay>
  );
};

Modal.propTypes = propTypes;
Modal.defaultProps = defaultProps;

export default Modal;
