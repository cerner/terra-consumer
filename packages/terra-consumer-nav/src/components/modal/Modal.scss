:local {
  .modal {
    background: var(--terra-consumer--modal-background, #fff);
    border-radius: 5px;
    display: block;
    left: 10px;
    overflow: auto;
    position: fixed;
    right: 10px;
    top: 60px;
    z-index: 300;
  }

  .modal-header {
    display: inline-block;
    left: 0;
    position: absolute;
    right: 0;
    text-align: center;
    top: 15px;
    width: 100%;
  }

  .modal-title {
    color: var(--terra-consumer--modal-title-color, #fff);
    font-size: 20px;
    line-height: 30px;
    text-align: center;
  }
  
  .overlay {
    background-color: var(--terra-consumer--overlay-background, #64696c);
    opacity: 1;
  }

  .close-icon {
    font-size: 16px;
  }
  
  .close-button {
    background: none;
    border: 0;
    color: var(--terra-consumer--modal-close-button-color, #fff);
    cursor: pointer;
    font-size: 16px;
    margin: 0;
    padding: 0;
    position: absolute;
    right: 15px;
    top: 5px;

    &:hover {
      color: var(--terra-consumer--modal-close-button-hover-color, #777);
    }
  }

}
