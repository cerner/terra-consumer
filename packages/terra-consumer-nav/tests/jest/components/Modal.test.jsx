import React from 'react';
import Modal from '../../../src/components/modal/Modal';


const modalContent = <div> This is a test content</div>;
const mockFn = jest.fn();

describe('Modal', () => {
  // Snapshot Tests
  it('should render a modal with title,close button and content', () => {
    const wrapper = shallow(<Modal
      title="Modal Tittle"
      content={modalContent}
      closeModal={mockFn}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
