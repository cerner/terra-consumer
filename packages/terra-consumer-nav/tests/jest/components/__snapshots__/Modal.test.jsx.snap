// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render a modal with title,close button and content 1`] = `
<Overlay
  backgroundStyle="light"
  className="overlay"
  content={
    <div>
       This is a test content
    </div>
  }
  isOpen={false}
  isRelativeToContainer={false}
  isScrollable={false}
>
  <div
    className="modal-title"
  >
    Modal Tittle
    <Button
      className="close-button"
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isReversed={false}
      onClick={[Function]}
      type="button"
      variant="default"
    >
      <IconClose
        data-name="Layer 1"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    </Button>
  </div>
  <div
    className="modal"
  />
</Overlay>
`;
